modules/timeutil/datetime.go

```diff
@@ -58,3 +59,29 @@ func DateTime(format string, datetime any) template.HTML {
 	}
 	panic(fmt.Sprintf("Unsupported format %s", format))
 }
+
+func ParseDateTimeGraceful(datetime any) (time.Time, error) {
+	switch val := datetime.(type) {
+	case string:
+		if timestamp, err := strconv.ParseInt(val, 10, 64); err == nil {
+			return ParseDateTimeGraceful(timestamp)
+		} else {
+			t, err := time.Parse(time.RFC3339, val)
+			if err != nil {
+				return time.Time{}, err
+			}
+			return t, nil
+		}
+	case int64:
+		switch {
+		case val > 946684800000000: // 2999-12-31 00:00:00 in milliseconds
+			return time.UnixMicro(val), nil
+		case val > 946645200000: // 2000-01-01 00:00:00 in milliseconds
+			return time.UnixMilli(val), nil
+		default:
+			return time.Unix(val, 0), nil
+		}
```

Comment:
I have objection for doing so. Keeping "guessing the unit" is quite fragile.

Either it strictly follows the Golang's time (int64 as nanosecond)

Or it strictly follows SI, use second as its unit, and use "0.001" for millisecond.